src/libs/ReportUtils.ts

```diff
@@ -1218,22 +1207,27 @@ function findLastAccessedReport(
 
     if (isFirstTimeNewExpensifyUser) {
         // Filter out the systemChat report from the reports list, as we don't want to drop the user into that report over Concierge when they first log in
-        sortedReports = sortedReports.filter((report) => !isSystemChat(report)) ?? [];
-        if (sortedReports.length === 1) {
-            return sortedReports[0];
+        filteredReports = filteredReports.filter((report) => !isSystemChat(report)) ?? [];
+        if (filteredReports.length === 1) {
+            return filteredReports[0];
         }
 
-        return adminReport ?? sortedReports.find((report) => !isConciergeChatReport(report));
+        return adminReport ?? filteredReports.find((report) => !isConciergeChatReport(report));
     }
 
     // If we only have two reports and one of them is the system chat, filter it out so we don't
     // overwrite showing the concierge chat
-    const hasSystemChat = sortedReports.find((report) => isSystemChat(report)) ?? false;
-    if (sortedReports.length === 2 && hasSystemChat) {
-        sortedReports = sortedReports.filter((report) => !isSystemChat(report)) ?? [];
+    const hasSystemChat = filteredReports.find((report) => isSystemChat(report)) ?? false;
+    if (filteredReports.length === 2 && hasSystemChat) {
+        filteredReports = filteredReports.filter((report) => !isSystemChat(report)) ?? [];
     }
 
-    return adminReport ?? sortedReports.at(-1);
+    // We are getting the last read report from the metadata of the report.
+    const lastRead = lodashMaxBy(filteredReports, (a) =>
+        new Date(reportMetadata?.[`${ONYXKEYS.COLLECTION.REPORT_METADATA}${a?.reportID}`]?.lastVisitTime ?? a?.lastReadTime ?? '').valueOf(),
```

Comment:
If we are 100% certain that the `lastReadTime` format will always be the same (`2024-07-08 08:55:01.665`), we can avoid parsing it into the `Date` object, as it is slow.

cc @mountiny

Reply:
However, if there is a chance that the format might change, it is safer to leave it as it is.

This change will probably have a minor performance boost, as the process is now `O(n)` and will only be called once, and only if the `reportID` is not in the URL.

Reply:
Constructing Date shouldn't have a big impact on the performance 🤔 Can you verify if it really affects the speed?

Reply:
Right now the difference is minor, but when we sorted the whole array, it had a significant impact overall.

| Comparison |  `main` | `branch` |
|--------|--------|--------|
| `Date` | 0.51 ms | 0.12 ms |  |
| `string` | 0.23 ms | 0.09 ms |
| Less time: | 55.41% | 25.37% |

Reply:
Let's verify if the format is always the same, it might be worth not using Date. Thank you for providing the benchmark 😄

Reply:
I think the format should be the same always, but I think we can make that change in a follow up PR just to make sure we can easily catch any regressions